core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java

```diff
@@ -155,8 +156,8 @@ private void initializeFromURL(final UrlRequestResult location) throws IOExcepti
         CacheEntry entry = new CacheEntry(resource.getLocation(), resource.getRequestVersion());
         entry.lock();
         try {
-            resource.setDownloadLocation(location.URL);
-            final URLConnection connection = ConnectionFactory.getConnectionFactory().openConnection(location.URL); // this won't change so should be okay not-synchronized
+            resource.setDownloadLocation(location.redirectUrl);
+            final URLConnection connection = ConnectionFactory.getConnectionFactory().openConnection(location.redirectUrl); // this won't change so should be okay not-synchronized
             connection.addRequestProperty("Accept-Encoding", "pack200-gzip, gzip");
 
             File localFile = CacheUtil.getCacheFile(resource.getLocation(), resource.getDownloadVersion());
```

Comment:
Line 164-165 condition is always false , maybe include this existing probable bug in the refactoring/cleanup
          Long size = location.length;
            if (size == null) {

Reply:
Yes, have not touched that method yet